docs/changing-an-existing-content-schema.md

```diff
@@ -0,0 +1,48 @@
+# Changing an existing content schema
+
+## General workflow
+
+* create a branch
+* make changes to the relevant schema fragment under `formats/<format name>.jsonnet`
+* change the relevant frontend examples (or add a new example)
+* run `rake` to compile a new schema.json and test the example against the schema
+* commit and push
+* open a PR
+* watch the multi-build status to see if all apps are compatible with the change
+* if the builds are successful, it can be merged
+
+## Adding a new field to a content schema
+
+Follow the General workflow described above. You are now free to add support for the field to apps. If you are adding a mandatory field you will need to additionally:
+
+* deploy the publishing app with changes to always populate the field
+* populate the field in any records inside content-store
+* follow the General workflow, adding the field to the `required` attribute
+* you can then update the frontend app to expect the field to always be present
+
+For example, imagine that you need to add a new optional field to the details hash of the `case_study` schema. The steps would be:
+
+1. edit [`formats/case_study.jsonnet`](/formats/case_study.jsonnet) to
+   add the new optional field
+1. run `rake`. This will:
+  1. regenerate the publisher [`schema.json`](/dist/formats/case_study/publisher/schema.json) to incorporate the changes you made to the `details`
+  1. regenerate the notification [`schema.json`](/dist/formats/case_study/notification/schema.json) to incorporate the changes
+  1. regenerate the frontend [`schema.json`](/dist/formats/case_study/frontend/schema.json) to incorporate the same changes
+  1. revalidate all example files to check if they are still valid after this change. This will pass, because the new field is optional
+1. [Optional step] you could add an additional example to illustrate how your new field should be used. You can add a new file in [examples/case_study/frontend](/examples/case_study/frontend)
+1. create a new branch and commit and push your changes
+  - this will run a branch build of govuk-content-schemas. This includes running the contract tests for each application which relies on the schemas. You'll get immediate feedback about whether publishing applications generate content items compatible with the new schema.
+1. once the tests pass, someone will merge your pull request and the new schemas will be available to use
```

Comment:
Style issue: numbering restarts for me here. In the past I've had to just number everything manually, as the auto-numbering doesn't always work.

Also, the bit about "someone will merge" is unnecessarily confusing - people should merge their own PRs. We could just lose these points as they're obvious.

Reply:
Fixed up the style issue, good spot.